Wrappers/Python/ccpi/viewer/ui/dialogs.py

```diff
@@ -137,6 +185,125 @@ def enableDisableDimZ(self):
             widgets['dim_Z_field'].setEnabled(True)
         else:
             widgets['dim_Z_field'].setEnabled(False)
+    
+    def preview(self):
+        pars = self.getRawAttrs()
+        
+        # retrieve info about image file from interface
+        dimensionality = [3, 2][self.getWidget('dimensionality').currentIndex()]
+        dimX, dimY, dimZ = pars['shape']
+        isFortran = pars['is_fortran']
+        isBigEndian = pars['is_big_endian']
+        # typecode = pars['typecode']
+        typecode = self.getWidget('dtype').currentText()
+
+        
+        if isFortran:
+            shape = (dimX, dimY)
+        else:
+            shape = (dimY, dimX)
+        if dimensionality == 3:
+            if isFortran:
+                shape = (dimX, dimY, dimZ)
+            else:
+                shape = (dimZ, dimY, dimX)
+
+
+        
+        
+        # Construct a data type
+        dt = np.dtype(typecode)
+        
+        if isBigEndian:
+            dt_txt = ">" # big endian
+        else:
+            dt_txt = "<"
+        dt = dt.newbyteorder(dt_txt)
+
+        bytes_per_element = dt.itemsize
+
+        # basic sanity check
+        file_size = os.stat(self.fname).st_size
+
+        expected_size = reduce (lambda x,y: x*y, shape, 1) * bytes_per_element        
+        
+        if file_size != expected_size:
+            errors = {"type": "size", 
+                      "file_size": file_size,
+                      "expected_size": expected_size}
+            return (errors)
```

Comment:
I think here an Exception should be raised.

I tested with a uint8 dataset.
If I chose the data type to be uint8 or int8, the preview dialog opened. If I chose anything else, the dialog did not open at all, and there was no error message. I think we should at least have some error dialog pop up.